src/crypto/DefaultSessionKeystore.h

```diff
@@ -0,0 +1,44 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#if CHIP_HAVE_CONFIG_H
+#include <crypto/CryptoBuildConfig.h>
+#endif
+
+#if CHIP_CRYPTO_PSA
+#include <crypto/PSASessionKeystore.h>
+#else
+#include <crypto/RawKeySessionKeystore.h>
+#endif
+
+namespace chip {
+namespace Crypto {
+
+// Define DefaultSessionKeystore type alias to reduce boilerplate code related to the fact that
+// when the PSA crypto backend is used, AES encryption/decryption function assume that the input
+// key handle carries a key reference instead of raw key material, so PSASessionKeystore must be
+// used instead of RawKeySessionKeystore to initialize the key handle.
+#if CHIP_CRYPTO_PSA
```

Comment:
The thing is that for PSA crypto the `RawKeySessionKeystore` is just wrong because PSA crypto assumes that `AesKeyHandle` is a key reference? Do you think that I should instead copy-paste the same `#ifdef .. #else` sequence in each of our sample apps? Wouldn't we just make people's lives harder without gaining any flexibility?

Reply:
I guess I expected the DefaultSessionKeystore not to appear by *value* anywhere (this is why SessionKeystore is an interface). The fact that it is not dependency-injected by pointer is what causes this type to even start existing. Overall, this showcases how our patterns are "rough around the edges".

I would argue that the only place this ifdef should  appear *at all* is in CommonCaseServerInitParams --> this is what has a lot of concrete types, which are then injected by pointer into Server class. In terms of controllers, they should be able to select their type beyond the ones here. The explicit appearance of DefaultSessionKeystore in the AndroidDeviceControllerWrapper is unfortunate, since there is no way to override which one will be used.

Overall, this shows up every time we try to add abstractions --> too much cut'n'paste and demo code that started without any "chosen" initialization and that relied on lots of magical defaults. This did not encourage careful choices or understanding of the choices.

What I would say is that the way this code is done, you have to *fork the SDK* to use a different version of SessionKeystore at a controller. This is problematic. Please see how OperationalKeystore interface is injected across the board.

For Java, we can start with RawKeySessionKeystore, but have a setter for the SessionKeystore configured after object init (since no call is make to SessionKeystore before operations take place).

Reply:
Discussed it on slack: I changed the controllers to use `RawKeySessionKeystore` if you prefer so. I kept the server to use `DefaultSessionKeystore` though to make it possible to build accessory apps with PSA crypto without making code changes.